### ISSUE TITLE
Account gets logged out after previously failing libsecret access

### ISSUE THREAD

**Issue by networkException** [element-hq/element-web#32121]
### Steps to reproduce

1. Start element-desktop on a new profile `element-desktop --profile-dir=/tmp/element-testing` while having a libsecret backend running and available for element to write session and seshat credentails
2. Log into an account
3. Close element-desktop <kbd>^C</kbd>
4. Open it again (this time blocking libsecret access which would happen over DBus) `firejail --noprofile --nodbus element-desktop --profile-dir=/tmp/element-testing`
5. Close element-desktop again <kbd>^C</kbd>
6. Open it again `element-desktop --profile-dir=/tmp/element-testing`

#### What did you expect?

I should stay logged into the account or at least view a cached version of it because credentials couldn't be retrieved from libsecret

#### What happened instead?

I see the login screen again, probably because libsecret failed the last time element-desktop started.

### Operating system

NixOS & Debian

### Application version

Element Desktop 1.11.53

### How did you install the app?

nixpkgs unstable & packages.element.io/debian

### Homeserver

matrix.org and selfhosted Synapse 1.98.0

### Will you send logs?

Yes (these were sent from the element running as `firejail --noprofile --nodbus element-desktop --profile-dir=/tmp/element-testing`, as per the steps above)

**Comment by t3chguy** [comment]
![image](https://github.com/element-hq/element-web/assets/2403652/e6f05290-4191-4c97-9627-4e2ad39d56be)

**Comment by networkException** [comment]
This might have fixed itself with recent changes to the ways secrets get stored (I don't want to risk my session trying).

I was personally also running into https://github.com/element-hq/element-web/issues/32041

**Comment by t3chguy** [comment]
Yes in theory safeStorage is more resilient to race conditions with keyrings due to it testing a write->read cycle to ensure the keyring is ready for use

**Comment by OdinVex** [comment]
Just to add, no, it still occurs. I still have to use my stub. Fussing about it being Arch or Arch-like isn't going to make the bug go away. If a secret wasn't fetched, it wasn't fetched, spin and wait for it or just notify and exit.
